### PR TITLE
[SPARK-39574][PS] Better error message when `ps.Index` is used for DataFrame/Series creation

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?
Better error message when `ps.Index` is used for DataFrame/Series creation.

### Why are the changes needed?
As part of [SPARK-39581](https://issues.apache.org/jira/browse/SPARK-39581).

The current error message is confusing and is from the pandas library. We should improve that.

### Does this PR introduce _any_ user-facing change?
Yes. 
**From:**
```py
>>> ps.DataFrame([1, 2], index=ps.Index([1, 2]))
Traceback (most recent call last):
...
ValueError: The truth value of a Int64Index is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().

>>> ps.Series([1, 2], index=ps.Index([1, 2]))
Traceback (most recent call last):
...
ValueError: The truth value of a Int64Index is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

**To:**
```py
>>> ps.DataFrame([1, 2], index=ps.Index([1, 2]))
Traceback (most recent call last):
...
TypeError: The given index cannot be a pandas-on-Spark index. Try pandas index or array-like.

>>> ps.Series([1, 2], index=ps.Index([1, 2]))
Traceback (most recent call last):
...
TypeError: The given index cannot be a pandas-on-Spark index. Try pandas index or array-like.
```

### How was this patch tested?
Unit tests.

### CODE REVIEW

**Comment by xinrong-databricks** [comment]
May I get a review @ueshin @HyukjinKwon @itholic @zhengruifeng @Yikun Thanks!

File: python/pyspark/pandas/tests/test_series.py [hunk 1]
```diff
@@ -54,6 +54,13 @@ def pser(self):
     def psser(self):
         return ps.from_pandas(self.pser)
 
+    def test_creation_index(self):
+        with self.assertRaisesRegex(
+            TypeError,
+            "The given index cannot be a pandas-on-Spark index. Try pandas.Index or array-like.",
+        ):
+            ps.Series([1, 2], index=ps.Index([1, 2]))
```
Comment (Yikun) [inline python/pyspark/pandas/tests/test_series.py:62]:
```
ps.Series([1, 2, 3],index=ps.MultiIndex.from_tuples([("a", "x"), ("a", "y"), ("b", "z")]))
```

nit: Maybe we also want to test for sub class of ps.Index?

File: python/pyspark/pandas/tests/test_dataframe.py [hunk 2]
```diff
@@ -96,6 +96,13 @@ def test_dataframe(self):
         index_cols = pdf.columns[column_mask]
         self.assert_eq(psdf[index_cols], pdf[index_cols])
 
+    def test_creation_index(self):
+        with self.assertRaisesRegex(
+            TypeError,
+            "The given index cannot be a pandas-on-Spark index. Try pandas.Index or array-like.",
+        ):
+            ps.DataFrame([1, 2], index=ps.Index([1, 2]))
```
Comment (Yikun) [inline python/pyspark/pandas/tests/test_dataframe.py:104]: ditto

File: python/pyspark/pandas/series.py [hunk 3]
```diff
@@ -405,6 +405,14 @@ def __init__(  # type: ignore[no-untyped-def]
                 assert not fastpath
                 s = data
             else:
+                from pyspark.pandas.indexes.base import Index
+
+                if isinstance(index, Index):
+                    raise TypeError(
```
Comment (Yikun) [inline python/pyspark/pandas/series.py:411]: Before this, ps raise the `ValueError` (same behavior with pandas, see below example).
```
pd.Series([1, 2, 3],index=ps.MultiIndex.from_tuples([("a", "x"), ("a", "y"), ("b", "z")]))
ValueError: The truth value of a MultiIndex is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
```

It's a user face behavior changes. But I think it's reasonable, we might want to mention this in migration guide?

File: python/pyspark/pandas/series.py [hunk 4]
```diff
@@ -405,6 +405,14 @@ def __init__(  # type: ignore[no-untyped-def]
...
+                        "The given index cannot be a pandas-on-Spark index. "
+                        "Try pandas.Index or array-like."
```
Comment (Yikun) [inline python/pyspark/pandas/series.py:413]:
```suggestion
                        "Try pandas index or array-like."
```

nit: Consider Index, MultiIndex, CategoricalIndex....so, maybe just

**Review by Yikun — COMMENTED** [review]
A reasonable improvement for me, Thanks!

**Review by HyukjinKwon — APPROVED** [review]
LGTM except the comments from @Yikun

File: python/pyspark/pandas/series.py (hunk 4 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py]: Good idea! Adjusted.

File: python/pyspark/pandas/tests/test_dataframe.py (hunk 2 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_dataframe.py:104]: Thanks!

File: python/pyspark/pandas/tests/test_series.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/tests/test_series.py:62]: Makes sense! Added tests.

File: python/pyspark/pandas/series.py (hunk 3 above)
Comment (xinrong-databricks) [reply python/pyspark/pandas/series.py:411]: That's a good question! I don't think pandas Series/DataFrame creation considers `ps.Index` as an input index officially. I have no objection if we want to mention this in the migration guide though.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**